backend/party/views.py

```diff
@@ -0,0 +1,41 @@
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from rest_framework import status
+from party.models import Party
+from reservations.models import Room
+from rest_framework import viewsets
+from party.serializers import PartySerializer
+
+class PartyViewSet(viewsets.ModelViewSet):
+    queryset = Party.objects.all()
+    serializer_class = PartySerializer
+    lookup_field = 'room_number'
+
+    def create(self, request, *args, **kwargs):
+        room_number = request.data['room_number']
+        secret = request.data['secret']
+        del request.data['secret']
+
+        # we allow looking up the room by
+        # * email
+        # * primary name
+        room = None
+        try:
+            room = Room.objects.get(name_hotel='Ballys', number=room_number)
+            if secret.lower() not in [room.guest.email.lower(), room.primary.lower()]:
+                return Response('is this really your room tho', status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This error isnt going through with a valid room number and invalid secret, instead goes to `this room is not real`

Reply:
See above. Spooky shit.